### PR TITLE
clojupyter: 0.3.2 -> 0.3.3

### PR DESCRIPTION
**Pull request by willcohen**
###### Description of changes

To address issues with clojupyter's instability, also noted at https://github.com/clojupyter/clojupyter/issues/162.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/jupyter-kernels/clojupyter/deps.nix [hunk 1]
```diff
@@ -1105,3 +1105,4 @@
 
   ];
   }
+
```
Comment (thiagokokada) [inline pkgs/applications/editors/jupyter-kernels/clojupyter/deps.nix:1108]: I think this extra blank line is breaking the Editor Config checks.

Comment (willcohen) [reply pkgs/applications/editors/jupyter-kernels/clojupyter/deps.nix:1108]: Good catch -- emacs config added that. Will revise.

Comment (thiagokokada) [reply pkgs/applications/editors/jupyter-kernels/clojupyter/deps.nix]: BTW, not really related to the review itself but did you generate this file by hand 🤔 ?

Comment (willcohen) [reply pkgs/applications/editors/jupyter-kernels/clojupyter/deps.nix]: I did not! I just ran `update.sh`, which seems like it just invokes `clj2nix`. I know there's been a lot of churn in different clojure nix tools but I figured no need to reopen that can of worms for just a version bump!

**Comment by thiagokokada** [comment]
Can you provide some instructions how to test this? I used Clojupyter in the past but not the Nix version, and trying to run the generated binary just cause the program to exit.

**Comment by willcohen** [comment]
I wish I had a more minimal example right now, but this is how I tested it:

Assuming nix-direnv is installed locally, and in a temp directory:

Contents of `.envrc`:
```
use flake
```

Contents of `flake.nix`:
```
{
  description = "Flake to manage project";

  inputs = {
    nixpkgs.url = "github:willcohen/nixpkgs/clojupyter-0.3.3";
    flake-utils.url = "github:numtide/flake-utils";
  };

  outputs = { self, nixpkgs, flake-utils }:
    flake-utils.lib.eachDefaultSystem (system:
    let
      pkgs = nixpkgs.legacyPackages.${system};
      jupyterWithClojupyter = pkgs.jupyter.override {
      definitions = {
        clojure = pkgs.clojupyter.definition;
      };
      };
    in {
      devShell = pkgs.mkShell {
        buildInputs = with pkgs; [
          jupyterWithClojupyter
        ];
      };
      devShells = {
        jdk = self.devShell;
      };
    });
}
```

In that directory, to switch to the local environment:
```
$ direnv allow
$ jupyter notebook
```

From there, jupyter notebook should start and generate a URL to be loaded, at which point a new notebook can be created or loaded with a Clojupyter kernel.

image

File: pkgs/applications/editors/jupyter-kernels/clojupyter/default.nix [hunk 2]
```diff
@@ -36,8 +36,8 @@ let
     src = fetchFromGitHub {
       owner = "clojupyter";
       repo = "clojupyter";
-      rev = "0.3.2";
-      sha256 = "1wphc7h74qlm9bcv5f95qhq1rq9gmcm5hvjblb01vffx996vr6jz";
+      rev = "0.3.3";
```
Comment (thiagokokada) [inline pkgs/applications/editors/jupyter-kernels/clojupyter/default.nix:39]:
```suggestion
      rev = version;
```

Comment (willcohen) [reply pkgs/applications/editors/jupyter-kernels/clojupyter/default.nix]: Also a good catch -- sorry for missing. Updated with latest force push.

**Review by thiagokokada — APPROVED** [review]

**Comment by willcohen** [comment]
Many thanks!

**Pull request merged**